CAS Content Nuke Neth Edition: Phase 1

TrollerNoob opened the pull request:
# About the pull request


https://github.com/user-attachments/assets/498f3750-a79e-4021-92fa-1944fda35e1e


This aims to not only finish some of Neth's planned CAS additions, but massively expands it with my own content. Also includes bugfixes, balance, QoL, new icons and immersive updates.

- 9 brand new equipment
  - 3 new Electronics
  - 3 new Crew Compartments
  - 2 new Weapons
  - 1 new Engine

- Anti-Air system added for CAS counterplay measures
  - Boilers and Queen gain the Skyspit ability, which damages dropship equipment when a Firemission strikes within its area and extinguishes mortar/dropship flares in the sky.
  - Neuro version of Skyspit creates Chaff instead, blocking camera vision and preventing all Direct Fire. Does not prevent Firemissions. Lasts longer than regular Skyspit.
  - Boilers in Extended Vision and Queen Eye are able to periodically see a motion detector-esque ping whenever CAS moves the camera near them while in flight.
  - Xenos gain a new structure called the Spore Pylon. Works the sameway Skyspit does, blocking Camera View and Direct Fire while inflicting damage during Firemissions. Lingers passively until destroyed. Can only be built on Hive Weeds.
  - Damaged equipment are corroded, preventing reloading while putting the equipment on a timer. If not repaired in 3 minutes, the equipment is destroyed. Does not prevent firing. Can have multiple corrosion on the same equipment, each requiring individual repairs.
  - Functionality added to the Dropship Handheld and Dropship Maintenance Computer.
  - Pilots, DCC or anyone with level 1 engineering or piloting must first scan the damaged equipment with the Tuner (dropship handheld), and then upload the scan to the Maintenance Computer in order to see the repair sequence. Or, you can guess if needed.
  - Each corrosion instance has a random sequence of 3 tools in order to repair them. You can not repair equipment while it's installed on the Dropship. 
  - Repair speed scales with Engineering, meaning DCC and MTs repair faster than Pilots. Work together if multiple equipment pieces are damaged.

 - Camera Console updated into the Auxiliary Console.
   - Complete with its own interface like the Weapon Console, the Auxiliary Console has access to all the support equipment on board the DS (and of course, the cameras). This lets the DCC handle things such as Fulton, Medevac and the other support equipment I've added.
   - Can not access any weapons nor fire missions.

 - Tactical Eyepiece added to Pilot's helmet (including Tex's).
   - Only works if you are **_inside_** the Dropship.
   - Gives the ability to see your Fire Mission strike zones.
   - Gives the ability to see your Direct Fire strike zones.
   - Gives the ability to see Antiair and Chaff danger zones.
   - Friendlies and Foes have an IFF HUD on them, allowing easier target acquisition. You do not see the IFF HUD of anyone inside the Dropship.

 - General Quality of Life improvements
    - Hula Girl now bobs and shakes depending on the dropship's movement, completing Monkey's original plan 4 years ago.
    - Weapon Console and Bellygun Console has their own unique HUD overlay for their cameras.
    - Grid preview option added to Firemission Creation so you can view how your firemission set up looks like
    - When the equipment is on c/d or the action can not be taken, the button will now greyed be out. For example, the FIRE button will be greyed when the weapon has no ammo or is on cooldown.
    - Ammo indicators change color as your ammo depletes, giving you a quicker heads up on when you're running low.
    - Quick mode added to Weapon Console's Direct Fire page. Lets you fire weapons at the press of a button, allowing for rapid firing of your entire arsenal without clunky menu navigation.
    -  Durability Reader added to Weapon Console's Firemission page. Gives information on what type of strike can penetrate the turf you're looking at, (green/yellow/red for all/firemissions only/nothing).
    - Fire Mission step inputs now save when you uninstall an equipment. You no longer have to reset the gimbals for each Fire Mission whenever you uninstall and reinstall the same equipment (as this will be happening more frequently now).
    - Added support for handheld dropship ammo (currently only for flare and bellygunner ammo)
    - Added support for topping off ammo already inside equipment when reloading
    - Dropship Fabricator now divides equipment and ammo based on what slot they can be installed on, so it's easy to find your Electronics, Weapons, etc.
    - Minimap now properly appears inside the window, and is live updated as it was before, giving pilots a better idea of how the groundside operation is going.
    - Fulton now alerts you when a fulton is launched while in flight.
    - Medevac now alerts you when a medevac stretcher is activated while in flight.
    - Paradrop system now has visual and audio indicators to better help marines know when it's time to jump.
    - Firemission names are now on the buttons instead of generic names like FM 1, FM 2, so you can quickly select the Firemission you need.
    - You can now input the X and Y coordinates manually in order to offset. This will help when camera is obstructed.
    - AN/AAQ-178 Weapon Targeting System's effects can now be toggled.

  - Balance Changes to existing ammo, generally decreasing explosive range but increasing explosive power
    - Cavebreaker var refactored (Chaff still prevents fire)
    - Metal Breaker var added, allows direct fire into Metal Roofs for certain ammo (Chaff still prevents fire)
    - SM '17' Fatty re-added as a Tier 2 purchasable Techweb. Audio and text cues re-added, visual green warning dot added to all cluster explosion locations.
    - AGM-98 'MOP' and the ATM-230D 'HELLHOUND IV', both cave piercing ammo, added as a Tier 2 purchasable Techweb. Each crate comes with 3 munitions. 
    - Laser Battery made more expensive (200 -> 250), has slightly less burn now (75 -> 60)
    - Antitank GAU fires more volleys, and fires them at a faster rate, given Metal Breaker (50 up from 40, 25% faster simulation speed)
    - Banshee now has a WP effect and 50 more explosive power (175 -> 225) but 20 more fall off (20 -> 40).
    - Keeper min and max accuracy range reduced, given Metal Breaker (made more accurate, so it lands closer to the lase)
    - Widowmaker travel time increased by 1 second (3 -> 4)
    - Napalm explosive power increased by 30%, and fall off doubled (200 -> 300, 25 -> 50)
    - Blue Dragon reworked, new icon, pull range lowered by 1, now has two small explosions, synced warning flash with other ammo, cost increased by 100 (300 -> 400)
    - Spotlight cost increased by 150 (50 -> 200)
    - Paradrop cost increased by 100 (50 -> 150)

  - Bug Fixes
    - CAS can no longer view into metal and higher roofs. They will still be able to see when executing a Firemission. Coordinate with your FTLs.
    - Fixed left electronic mount point on the Normandy being set to null
    - Fixed missing Fire Mission length input when creating Firemissions. Use lengths divisible into 12 for best results (1 2 4 6 12)
    - Fixed GAU/Laser ammo not crediting the owner in kill logs

# Explain why it's good for the game

This is an update aimed at making CAS much more interactive for both sides. Xenos now have an avenue for punishing CAS, preventing a one-way game between them and Pilots. They are also able to create areas that dissuade Pilots from firing, while not being as flat out restrictive as a Pylon. Boilers, specifically, are now specialized at not only being a mobile mortar, but a mobile anti-air battery. They have unique tools to detect and disrupt CAS.

Pilots gain a bunch of new equipment and immersive features, hopefully opening up the way for a lot more different combination for Dropship setups. Transport POs will also have more to do too.

DCC, with the addition of the Auxiliary Console, are now a proactive role, offloading a lot of pressure from Pilots as they'll be able to handle much of the busywork. They are also faster at repairs.

Most of all, this finally adds in the long promised and awaited features that Neth wanted to add many years ago. The general Antiair system and 4 of the new equipment are his ideas that were set to be added, but was never able to come to fruition. With the existing CAS system now modernized and fleshed out, this opens the gate for Phase 2 of Neth's plan, the Gunship Hover Mode.

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

New Equipment
<img width="853" height="183" alt="image" src="https://github.com/user-attachments/assets/6b70169f-dc3f-46e9-8997-a7e1284614cd" />

New Ammo
<img width="683" height="248" alt="image" src="https://github.com/user-attachments/assets/12d42f6a-f20b-4017-8978-16debf0b9a34" />

New Paradrop
<img width="125" height="108" alt="image" src="https://github.com/user-attachments/assets/64066c93-4baa-439f-ac0b-7099c3442fc4" />

New BLU-200 'Dragon's Breath'
<img width="181" height="86" alt="image" src="https://github.com/user-attachments/assets/5d780915-0aa5-4a4b-9297-b196e404cd6c" />

New Maintenance Tools Off/On
<img width="161" height="98" alt="image" src="https://github.com/user-attachments/assets/872ed0e7-3381-4bd6-9e7c-8002ff5b27aa" />
<img width="161" height="99" alt="image" src="https://github.com/user-attachments/assets/062814ea-cc99-4d33-be7f-1cac6634e7de" />

Bellygunner and ammo
<img width="88" height="177" alt="image" src="https://github.com/user-attachments/assets/7d00b3a8-f826-451c-9b41-9485b48eb8c3" />

<hr>

Weapon Console Panel
<img width="1411" height="653" alt="image" src="https://github.com/user-attachments/assets/6228defc-d0f6-43a8-8cd1-0c8cbb9093d1" />

Camera Console Panel
<img width="1410" height="695" alt="image" src="https://github.com/user-attachments/assets/025f6517-e1ab-455d-abc4-8548dccb5f0d" />

Bellygunner Console Panel
<img width="1413" height="660" alt="image" src="https://github.com/user-attachments/assets/64bca5c7-9a64-41d1-a0ac-cea5904f1b9b" />

<hr>

Live Demonstrations

IFF Target Painting

https://github.com/user-attachments/assets/056c88ac-cc80-499e-b2c5-248257909a05

IFF Direct Fire

https://github.com/user-attachments/assets/8e2f9e7f-3cff-4699-8878-9c59144d5f11

IFF Fire Missions

https://github.com/user-attachments/assets/e9c378b7-84e8-44fe-af3a-ff465e8392bc

Fatty

https://github.com/user-attachments/assets/a0f4a10f-b1db-4015-ae6b-7d5577e94bf8

Quick Mode Direct Fire

https://github.com/user-attachments/assets/f339d341-2ba2-4b7e-8833-04b9b96a5b77

Grid Firemission Preview
<img width="954" height="967" alt="image" src="https://github.com/user-attachments/assets/7c67f510-5e87-4d0d-bbc7-50874dac052b" />

Tactical Map
<img width="2105" height="983" alt="image" src="https://github.com/user-attachments/assets/c1ecbaa5-aa1b-492d-a3df-4d26cabc7d1b" />

<hr>

Boiler Psychic Pings

https://github.com/user-attachments/assets/9597f837-4215-410a-9929-a9da22c85ea7

Boiler Antiair Success

https://github.com/user-attachments/assets/a4b75a13-24b5-432f-b587-d8b979e4c676

Pilot Antiair Failure

https://github.com/user-attachments/assets/386dbcec-1434-4f8b-87e5-d14072e7d5df

Maintenance Diagnosis

https://github.com/user-attachments/assets/c05d9046-a48e-4cf8-b9a6-e012152c485f

<hr>
Equipment Demonstrations

Rappel Deployment Hatch

https://github.com/user-attachments/assets/d2c7128f-d9d9-4da5-b24c-42c0db7b6606

Rappel Deployment Ground

https://github.com/user-attachments/assets/1785d793-8336-482c-8700-3f5b6cc6fd64

Rappel Hostile Boarding

https://github.com/user-attachments/assets/0d7707d3-2b42-4b1c-9dc9-5c218a6959f6

Rappel Boarding Warning

https://github.com/user-attachments/assets/24346da0-ea80-4d19-bcbf-b676add58fbb

Paradrop Display System

https://github.com/user-attachments/assets/67846c03-02b6-4a0c-88e0-9f06be3b5b45

Autoreloader System

https://github.com/user-attachments/assets/157acc4c-65fd-4371-baf8-46b093bf143a

Bomb Bay System

https://github.com/user-attachments/assets/839c93f5-78e7-4e4a-ab40-2e36eacba2ae

Bellygunner System

https://github.com/user-attachments/assets/a995eaeb-515c-43e4-aa3c-e5de660c11cc

<hr>

Spore Pylon

<img width="361" height="363" alt="image" src="https://github.com/user-attachments/assets/93442720-8993-4f79-b38c-07d79b69f2d5" />

Skyspit
<img width="133" height="132" alt="image" src="https://github.com/user-attachments/assets/861870b5-2221-4ccd-8033-62282c6c9a42" />

Skyspit Acid

https://github.com/user-attachments/assets/31c13cc7-705f-4420-82e2-6626f8275b2a

Skyspit Neuro

https://github.com/user-attachments/assets/d81a2eb5-d8e1-40c8-9c90-b258d9400895

Skyspit Extinguishing Sky Flares

https://github.com/user-attachments/assets/022ffad3-9c59-47a5-8142-febcaa9ecaa5

</details>


# Changelog
:cl:
add: The last trip to Chinook 91 GSO station has seen a massive overhaul to the Almayer's Aerospace fleet!
add: It seems there has been reports of certain mutations among the XX-121...
add: Hula Girl now shakes and bobs as the Alamo flies!
add: Added a falling shell effect for most CAS ammo types.
add: Added the Pilot Optic to all pilot helmets, a tactical eyepiece that gives the ability to see IFF HUDs on friends and foe, firemission and direct fire impact zones and dangerous antiair or chaff zones.
add: Updated the Dropship's Camera Console into the Auxiliary Support Console, giving it much of the same functionality as the Weapon Console without the ability to access weapons, direct fire or fire missions.
add: The infamous SM-17 'Fatty' has been reintroduced to fleets in limited capacity after thorough investigations. It is now a Tier 2 Tech.
add: Added a new Dropship Weapon, the MK. 14 Missile Silo, which can only be used in firemissions.
add: Added four new Missile ammo types which spawns shrapnel, MK.12 'Zeus', MK.91 'SGW', MK.25 'Banshee' and ATM-230D 'HELLHOUND IV'.
add: Added a new Dropship Weapon, the LAB-107 Bomb Bay, which can only be used for direct fires and fits on a Crew Compartment slot.
add: Added three new Bomb ammo types which have an audio cue, SM-21 'Chubby', AGM-81 'Firecracker', AGM-98 'MOP'.
add: Added a new Dropship Crew, the HRU-3 Rappel Deployment System, allowing two way deployment between the Dropship and Ground but requires a Rappel Harness on the belt. Beware, smaller xenos can climb the rope without the need of one.
add: Added a new Dropship Crew, the GAU-24/B Belly Gunner Station System, allowing DCC to give additional supporting direct fire.
add: Added one new exclusive Heavygun ammo, PGU-110 25mm Compact ammo crate, capable of being loaded by hand.
add: Added the unused flare Minirocket variant, the AGR-59-F, which deals less damage but spreads a small amount of starshell ash in the area.
add: Added a new Dropship Crew, the RMT-08 Autoreloader System, stores up to three missiles or rockets for mid-flight reloading. Must be manually winched to complete the reload with a lengthy cooldown.
add: Added a new Electronic, the AN/ALE-557 Flare Launcher, providing battlefield illumination for ground forces.
add: Added one new Flare ammo type, AN/ALE-557 Flare Launcher cartridge, capable of being loaded by hand.
add: Added a new Electronic, the AN/ASQ-183 Target Designation Pod, allowing the ability to offset direct fire up to 3 tiles away from the lase. A yellow signal dot appears at your new offsetted location, visible to everyone on the ground.
add: Added a new Electronic, the TF-107/B-01 Vertical Exhaust Nozzle, increasing how fast you fire during your firemission by 33%, but decreasing direct fire accuracy by 3 and increasing ammo travel time by 2 seconds.
add: Reworked the AN/LEN-15 Spotlight, it now provides a medium sized spotlight onto the selected signal when in the air and turned on.
add: Added a new Engine, the TF-220/A-14 ramrocket, increasing maximum firemission length by 4, decreasing delay between initiating a fire mission by 50%, but decreasing flyby time and increasing transport time by 25%.
add: Anti-Air backend support now added. Boilers and Queens gain access to the Skyspit ability, which can damage dropships and extinguish flares in the sky. Boilers must be in extended vision to use Skyspit, Queen does not.
add: Boiler's Neurotoxic Skyspit creates Chaff instead, blocking CAS vision and direct fire. Does not prevent Firemissions.
add: Extended Sight Boilers and Queen's Eye can now detect periodic psychic pings whenever the CAS camera moves near them while in flight.
add: Added Spore Pylon, which works similar to Skyspit but is persistent until the structure is destroyed. Blocks direct fire and camera view, inflicts equipment damage during firemissions. Can only be built on hive weeds.
add: Added ability for Dropship Equipment to be damaged. When executing a firemission into a Skyspit marked zone, weapon equipment is damaged. Multiple stacks on one equipment are possible. Direct fire is unaffected.
add: Damaged equipment are corroded, can not be reloaded and will be destroyed in 3 minutes if not repaired.
add: Two new tools added, the Aircraft Maintenance Tuner and the Dropship Maintenance Computer.
add: Pilots or DCC must first scan the damaged equipment with the Tuner, and then upload it into the Computer to reveal the correct repair steps. Guessing also works.
add: Damaged equipment can not be repaired while installed, and repair speed scales with Engineering. DCC repair faster than POs as a result.
add: Extra Tuners or Computers can be bought for points at the vendor.
balance: SM-17 'Fatty's cluster explosions now have green warning dot indicators.
balance: SM-17 'Fatty', AGM-98 'MOP' and the ATM-230D 'HELLHOUND IV' can only be bought via Tier 2 Techweb. Each crate carries only three munitions, use them wisely.
balance: BTU-17/LW Hi-Cap Laser Battery cost increased from 200 to 250, burn decreased from 75 to 60.
balance: PGU-105 30mm Anti-tank ammo crate fires volleys 25% faster and now fires 50 times per volley, up from 40. It can now pierce metal roofs on direct fire.
balance: GBU-67 'Keeper II' min and max accuracy decreased to 2 and 3 tiles, making it much more accurate. It can now pierce metal roofs on direct fire.
balance: AGM-228W 'Banshee' given a new name, now has a White Phosphorus effect. Explosive power increased from 175 -> 225 but falloff increased from 20 to 40.
balance: AIM-224B 'Widowmaker' travel time increased from 3 -> 4 seconds.
balance: BLU-200 'Dragon's Breath' pull range lowered by 1, now has 2 miniature explosions, cost increased from 300 -> 400
balance: AGM-99 'Napalm' explosion power increased from 200 to 300, falloff increased from 25 to 50.
balance: AN/LEN-15 Spotlight cost increased from 50 -> 200.
balance: HPU-1 Paradrop Deployment System cost increased from 50 -> 150.
qol: Firemission input steps now save when uninstalling equipment. You no longer have to re-input them each time.
qol: You can now refill Minirocket and Missile Silo ammo while they're installed without having to unload the ammo. 
qol: UI Action buttons now grey themselves out when the action can not be taken.
qol: Added ceiling reader to the TARGET panel, giving information on what type of strikes will work for the ceiling level.
qol: Dropship Part Fabricator now categorizes equipment and ammo by where they can be installed.
qol: Quick mode added to Direct Fire panel, displaying all weapons as buttons on the left and right. Simply press the weapon's button to fire it at the currently selected target.
qol: Grid preview option added to Firemission creation, allowing pilots to preview how their Firemission setup looks like.
qol: Fulton now has an audible and text cue whenever a fulton is launched while in flight.
qol: Medevac now has an audible and text cue whenever a stretcher is activated while in flight.
qol: Paradrop now has audible and visual cues to better let marines know when it's time to jump.
qol: You can now manually input the X and Y when offsetting.
qol: AN/AAQ-178 Weapon Targeting System can now be toggled on and off while mid-flight, letting you forgo the effects tactically.
qol: Probably a bunch of other stuff I forgot.
ui: Dropship Weapon Console and Bellygun Console cameras given their own unique overlays.
ui: Camera Console and Bellygun Console have their own unique color schemes.
imageadd: Added projectile icon states for each ship ammo type.
imageadd: Added new icon states for the HPU-1 Paradrop Deployment System.
imageadd: Added new icon states for BLU-200 'Dragon's Breath'.
imageadd: Added icons for the 9 new equipment pieces, their ammo, the dropship maintenance tools, spore pylon, CAS IFF HUD effects, xeno antiair/chaff effects.
fix: Fixed CAS Camera being able to see into Metal and stronger roofs. You'll be able to see your Firemission play out still.
fix: Fixed missing Firemission Length input during Firemission Creation.
fix: Fixed null mount point on left electronic slot for the Normandy.
fix: Fixed Laser and GAU ammo not properly recording the owner for kills
fix: Minimap now properly displays inside the console window, and is live updated.
refactor: Refactored direct fire code to properly update the impact area when the signal is lost mid-transit.
refactor: Refactored fabricator code to get each equipment's mount point in order to categorize them.
refactor: Refactored Minirocket code to use the modern global proc ref
refactor: Refactored GAU simulation code, now allows looping values less than 1 second
code: Changed Cavebreaker from being an Equipment var to an Ammo var, added a Metal Breaker var for metal roofs.
code: Updated rappel_rope.dm file to work with the rappel_system.
maptweak: Added pre-existing Aircraft Maintenance Tuners and Dropship Maintenance Computers near both dropships, complete with all the tools they'd need for repairs.
maptweak: Added some pre-existing Rappel Harnesses near the Dropship Parts Fabricator.
maptweak: Added some of the goodies gotten from Chinook Station to the Pilot Dorms and Dropship Control Bubble.
add: Credits: NethIafin for the original ideas, Spheretech for the Autoreloader and Ram Rocket sprites.
/:cl:

TrollerNoob committed:
Content Nuke

cm13-github commented:
This pull request has conflicts, please resolve those before we can evaluate the pull request.

Detective-Google commented on code/datums/mob_hud.dm:
```diff
@@ -173,6 +174,26 @@ GLOBAL_LIST_INIT_TYPED(huds, /datum/mob_hud, list(
+	if(user.client && isobserver(user))
+		refresh_xeno_telegraphs_for_client(user.client)
+
+/datum/mob_hud/xeno/proc/refresh_xeno_telegraphs_for_client(client/C)
```
client/C -> client/client or client/xeno_client, etc

NovaCypher1 commented:
I love a bit of PO love, this looks awesome!

Detective-Google commented on code/modules/mob/living/carbon/xenomorph/castes/Boiler.dm:
```diff
@@ -61,6 +61,7 @@
 	weed_food_states_flipped = list("Boiler_1","Boiler_2","Boiler_3")
 
 	var/datum/effect_system/smoke_spread/xeno_acid/smoke
+	var/last_psychic_pulse_time = 0
```
add a comment for what this variable is for e.g. 

```suggestion
    /// Last time a dropship camera detection pulse was triggered
	var/last_psychic_pulse_time = 0
```

consider using COOLDOWN_DECLARE (basically just a pretty formatting for this kind of world.time cooldown timer.)

Detective-Google commented on code/modules/cm_marines/dropship_equipment.dm:
```diff
@@ -21,7 +20,13 @@
 	var/skill_required = SKILL_PILOT_TRAINED
 	var/combat_equipment = TRUE
 	var/faction_exclusive //if null all factions can print it
+	var/stackable_ammo = FALSE 	/// Whether the ammo inside this equipment can be directly replenished without needing to uninstall the existing ammo
```
https://github.com/cmss13-devs/cmss13/blob/master/.github/guides/AUTODOC.md#required 

these vars (really the more the merrier but especially these) should be autodoc'd

Detective-Google commented on code/modules/cm_marines/dropship_equipment.dm:
```diff
@@ -51,34 +56,305 @@
+				return TRUE
+			//need the welder to be active for welding repairs
+			if(next_step == "WELDER" && istype(I, /obj/item/tool/weldingtool))
+				var/obj/item/tool/weldingtool/WELD = I
+				if(!WELD.welding)
+					to_chat(user, SPAN_WARNING("The welder must be activated!"))
+					effect.repairing = FALSE
+					return TRUE
+				WELD.eyecheck(user)
+			// Play sounds and effects
```
```suggestion
				var/obj/item/tool/weldingtool/weld = I
				if(!weld.welding)
					to_chat(user, SPAN_WARNING("The welder must be activated!"))
					effect.repairing = FALSE
					return TRUE
				weld.eyecheck(user)
			// Play sounds and effects
```

TrollerNoob committed:
1 letter var elimination

TrollerNoob committed:
Autodocing vars

TrollerNoob committed:
Merge branch 'master' of https://github.com/TrollerNoob/cmss13 into CAS-Content-Nuke--Phase-1

cm13-github commented:
Conflicts have been resolved. A maintainer will review the pull request shortly.

TrollerNoob committed:
Fixed scaling of large racks, fixed huds/helmet_visors merge conflict

TrollerNoob committed:
Fixing map runtime due to missing ;

TrollerNoob committed:
Fixed indentations

TrollerNoob committed:
maintenance repair tools converted into a helper

TrollerNoob committed:
Fixed empty block

TrollerNoob committed:
Prettier formatting fix

cuberound commented:
ramjet does not work as you say, the grace period does not work like that, you have reduced the time till the sound is played by half but all the other times are unrelated and independend of eachother, so the sound gets played sooner but the first impact is still after 10 seconds

cuberound commented:
with the gau ammo the firing speed is unchanged, I think that you can not sleep for less then a tick

cuberound commented:
(there is stuff that got balance denied in past, as sped up FMs and cave piercing ammo, but opinions on it might have changed)

cuberound commented:
you are also, I assume unintentionaly, reverting bugfixes from this pr https://github.com/cmss13-devs/cmss13/pull/9938

TrollerNoob committed:
Ram Rocket firemission sync and GAU simulation speed fixes

TrollerNoob committed:
Revert unused omni sentry price change

TrollerNoob committed:
1 letter var kill, more AUTODOCing, increased autoreloader price to 500 from 300

TrollerNoob committed:
added a marine tip

cm13-github commented:
This pull request has conflicts, please resolve those before we can evaluate the pull request.

TrollerNoob committed:
Merge branch 'master' into CAS-Content-Nuke--Phase-1

cm13-github commented:
Conflicts have been resolved. A maintainer will review the pull request shortly.

TrollerNoob committed:
Merge branch 'master' into CAS-Content-Nuke--Phase-1

cm13-github commented:
Conflicts have been resolved. A maintainer will review the pull request shortly.

TrollerNoob committed:
force returning update_icon for the spotlight beam

TrollerNoob committed:
Merge branch 'CAS-Content-Nuke--Phase-1' of https://github.com/TrollerNoob/cmss13 into CAS-Content-Nuke--Phase-1

TrollerNoob committed:
trying to fix unit test GC infinite loop

TrollerNoob committed:
Fixed protection_flags error

TrollerNoob committed:
1 letter var kill, flare ammo inhand icon tweaks, observers now see xeno antiair overlays, flare/paradrop cost changes

TrollerNoob committed:
BLU-200 'Dragon's Breath' reworked, new icon, cost increased from 300 -> 400, autoreloader and target designation pod bugfix

TrollerNoob committed:
Bunker Buster impact icon tweaks

TrollerNoob committed:
Camera Console no longer detects bellygun

TrollerNoob committed:
Bellygun console now auto spawns when installing the bellygun

TrollerNoob committed:
Hellhound step delay increased

TrollerNoob committed:
Increased Rappel Harness item size

TrollerNoob committed:
Added vsfx to damaged dropship equipment

TrollerNoob committed:
Bellygun Ammo reduced to 300, shots fired increased to 30

TrollerNoob committed:
Removes in worlds, simplified maintenance tool list to jusr use their type paths, clarified marine/pilot distinction comment for iff hud, attempts to fix wonky master merge

TrollerNoob committed:
Merge branch 'master' of https://github.com/cmss13-devs/cmss13 into CAS-Content-Nuke--Phase-1

cm13-github commented:
This pull request has conflicts, please resolve those before we can evaluate the pull request.

TrollerNoob committed:
Merge branch 'master' of https://github.com/cmss13-devs/cmss13 into CAS-Content-Nuke--Phase-1

cm13-github commented:
Conflicts have been resolved. A maintainer will review the pull request shortly.

TrollerNoob committed:
Merge branch 'master' into CAS-Content-Nuke--Phase-1

Nanu308 reviewed (dismissed):
Mapping approved

rodcowartt commented:
Hey @TrollerNoob hi there! I love this content nuke on CAS and really want to try it out, so I was wondering: can I port it to another build, fork of this cmss13 (https://github.com/ss220club/BandaMarines)? Seeing it stale in weeks is really sad to me! You can make pull yourself if you want to; if no, please let me know; I'll do it myself, authoring you, of course (no pressure)! Anyways thanks for the content and love your vids <3!

TrollerNoob committed:
Merge branch 'master' of https://github.com/cmss13-devs/cmss13 into CAS-Content-Nuke--Phase-1

TrollerNoob committed:
Fixed being able to use multiple heavygun consoles simultaneously, updated camera/heavygun minimap to the current one

TrollerNoob committed:
fixed duplicate definitions

TrollerNoob committed:
Updated pilot optic button icon to be in line with the current hud

TrollerNoob committed:
re-organized icons in dropship_ammo.dmi, touched up fatty and napalm icons, added in unused minirocket flare variant

TrollerNoob committed:
ammo is now categorized in the fabricator, optimized dropship fabricator categorization

TrollerNoob committed:
fixed missing audio cues from repair tools

TrollerNoob committed:
fixed missing toolbox, maintenance computers, welder goggles and oil splatters

TrollerNoob committed:
Fixup maps in TGM format

303883eeabef: maps/map_files/USS_Almayer/USS_Almayer.dmm

Automatically commited by: tools\mapmerge2\fixup.py

TrollerNoob committed:
linters fix

TrollerNoob committed:
updated bunker buster impact icon, added additional warning audio cues

cm13-github commented:
This pull request has conflicts, please resolve those before we can evaluate the pull request.

TrollerNoob committed:
Added projectile impact animations and icons for all ammo types, similar to CAS/OB. Polished minirocket sprites, tweaked Banshee phosp spawn

TrollerNoob committed:
Merge branch 'master' of https://github.com/cmss13-devs/cmss13 into CAS-Content-Nuke--Phase-1

cm13-github commented:
Conflicts have been resolved. A maintainer will review the pull request shortly.

TrollerNoob committed:
fixed cas missile overlay being rotated the wrong way, slightly reduced missile size

TrollerNoob committed:
Moved cave piercing ammo into Tier 2 tech, polished cluster projectile icons, unified ammo descriptions

TrollerNoob committed:
added back fatty as a tier 2 tech with audio + text warnings, cluster explosions have green warning dots

TrollerNoob committed:
Made the tac map appear inside the Mfd Panel instead of layering over your screen, added hand unloading for applicable dropship equipment

TrollerNoob committed:
linters fixed

TrollerNoob committed:
null check added to tacmap creation

TrollerNoob committed:
additional params added to minimap error

TrollerNoob committed:
linters fix

TrollerNoob committed:
buffed spore tower to have both anti-firemission and anti-directfire effects

TrollerNoob committed:
global code clean up, 1 letter var kill, icon files re-organized

TrollerNoob committed:
Added Quick mode to direct fire, letting pilots have access to direct firing all 4 weapons in one panel

TrollerNoob committed:
Added an audible and text alert to the Fulton system whenever a new fulton is detected while in flight

TrollerNoob committed:
added a grid preview to firemission creation, general QoL improvements like firemission names appearing on buttons, scrolling, etc

TrollerNoob committed:
fixed tac map and firemission bug

TrollerNoob committed:
missile shrapnel count reduced, skyspit var name changed to be in line with the other boiler related abilities

TrollerNoob committed:
red outline added to a variety of buttons to indicate when toggled, bugfixes

TrollerNoob committed:
audio and text cues added to medevac, camera console renamed to auxiliary support console, more bugfixes

TrollerNoob committed:
lowered max antiair pylon count by 1, clarified description

TrollerNoob committed:
targeting system's effects are now toggleable

TrollerNoob committed:
added manual number style inputting for offsetting

TrollerNoob committed:
more code polishing and 1 letter var removals

TrollerNoob committed:
fixed grid orientation to face north

Detective-Google commented on code/game/camera_manager/camera_manager.dm:
```diff
@@ -62,6 +63,38 @@
+					var/atom/movable/screen/fullscreen/pilot_camera/typed_overlay = overlay
+					plane:vis_contents -= typed_overlay
+					qdel(typed_overlay)
+				plane:vars["cas_hud_overlays"] = list()
```
https://github.com/cmss13-devs/cmss13/blob/master/.github/guides/STANDARDS.md#dont-override-type-safety-checks

Detective-Google reviewed (changes requested):
in general I think you're going to end up needing to break this PR up into components that are actually reviewable; there's just too much going on here to really comprehend all of it at once; 12k lines over 100 different files just isn't gonna happen.

TrollerNoob committed:
removed type safety check overrides

TrollerNoob commented:
> in general I think you're going to end up needing to break this PR up into components that are actually reviewable; there's just too much going on here to really comprehend all of it at once; 12k lines over 100 different files just isn't gonna happen.

Alright, I'll atomize the PR into two parts. The first will be just the antiair mechanic + QoL features, and the second will be all the new equipment/ammo/etc.

Detective-Google commented:
closed in favor of atomized pieces
